datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -402,25 +404,50 @@ def in_developer_mode(self):
         self._log_deprecation('in_developer_mode')
         return False
 
+    def find_typos_in_options(self, user_config, models_config, level):
+        user_config = user_config or {}
+        models_config = models_config or {}
+        typos = []  # type: List[str]
+
+        known_options = dict(models_config).keys()
+        unknown_options = sorted(list(user_config.keys() - known_options))
+        for unknown_option in unknown_options:
+            most_similar_known_option = (None, 0)
+            for known_option in known_options:
+                ratio = jaro_distance(unknown_option, known_option)
+                if ratio > most_similar_known_option[1]:
+                    most_similar_known_option = (known_option, ratio)
+            if most_similar_known_option[1] >= TYPO_SIMILARITY_THRESHOLD:
+                typos = typos + [unknown_option]
+                message = (
+                    "Detected potential typo in configuration option in {}/{} section: {}. Did you mean `{}`?"
+                ).format(self.name, level, unknown_option, most_similar_known_option[0])
+                self.log.warning(message)
+        return typos
+
     def load_configuration_models(self, package_path=None):
         if package_path is None:
             # 'datadog_checks.<PACKAGE>.<MODULE>...'
             module_parts = self.__module__.split('.')
             package_path = '{}.config_models'.format('.'.join(module_parts[:2]))
-
         if self._config_model_shared is None:
             raw_shared_config = self._get_config_model_initialization_data()
-            raw_shared_config.update(self._get_shared_config())
+            intg_shared_config = self._get_shared_config()
+            raw_shared_config.update(intg_shared_config)
 
             shared_config = self.load_configuration_model(package_path, 'SharedConfig', raw_shared_config)
```

Comment:
Does `shared_config` contain all of the possible known options that a particular check can have? Would that also include any hidden options? For example if a customer uses Openmetrics V1 option `prometheus_url` on a check with only Openmetrics V2 options visible.

Reply:
It includes hidden options like `prometheus_url`, but not options not linked to the integration's spec. For example, if a check were only openmetrics v2 and didn't have a hidden legacy openmentrics template, `prometheus_url` would not be included here.

Reply:
Hm I'm not sure how important this is, but I wonder if there are any hidden configs that should stay hidden from customers and be ignored from the typo checking.

Reply:
There are not many cases of hidden configs, but that is a good question. But I think that if a user knows to type something like `prometheus_rl` then it's pretty likely they are aware of and want to set `prometheus_url`, and the warning log would be useful to them.

Reply:
Updated to include multiple candidates!